lib/correios_sigep/logistic_reverse/collect_number_actions/process_invalid_request.rb

```diff
@@ -23,9 +20,26 @@ def errors
             Models::CorreiosResponseCodes::INEXISTENT_ZIPCODE => Models::Errors::InexistentZipcode,
             Models::CorreiosResponseCodes::UNAVAILABLE_HOUSE_COLLECT => Models::Errors::UnavailableHouseCollect,
             Models::CorreiosResponseCodes::COLLECT_NOT_ANSWERED_FOR_THE_ZIPCODE => Models::Errors::CollectNotAnsweredForTheZipcode,
-            Models::CorreiosResponseCodes::NOT_CONFIGURED_CLIENT => Models::Errors::NotConfiguredClient
+            Models::CorreiosResponseCodes::NOT_CONFIGURED_CLIENT => Models::Errors::NotConfiguredClient,
+            Models::CorreiosResponseCodes::INVALID_CONTRACT => Models::Errors::InvalidContract,
+            Models::CorreiosResponseCodes::INVALID_DECLARED_VALUE => Models::Errors::InvalidDeclaredValue
           }
         end
+
+        def error_code
+          return nil if @response.search('//cod_erro').empty?
+          @response.search('//cod_erro').text.to_i
+        end
+
+        def new_error_code
+          return nil if @response.search('//codigo_erro').empty?
+          @response.search('//codigo_erro').text.to_i
+        end
+
+        def error_message
+          error_message = @response.search('//msg_erro').text
+          @response.search('//descricao_erro').text if error_message.blank?
```

Comment:
one last thing. I think if `msg_erro`  is not empty, and `descricao_erro` is empty, it will failt. 

Something like:

```
error_message = 'return value'
'another return' if error_message.blank?
```

it will never return `error_message`.

Maybe changing it to:
```
@response.search('//msg_erro').text) ||  @response.search('//descricao_erro').text
```